File: Python3/프로그래머스/1/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.py
def solution(bandage, health, attacks):
    t, x, y = bandage
    full = health
    time, rest = 0, 0
    for attt, attn in attacks:
        rest = attt - time - 1
        health = min(health + (x * rest) + int(rest / t) * y, full) - attn
        time = attt
        # print("현재 시각", time, "에 몬스터에게", attn, "만큼 공격 당해 현재 체력", health)
        if health <= 0: return -1
        
    return health

File: C++/프로그래머스/1/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> bandage, int health, vector<vector<int>> attacks) {
    int answer = health;
    int time = 0, rest = 0;
    
    for(vector<int> attack : attacks) {
        rest = attack[0] - time - 1;
        answer = min(answer + (bandage[1] * rest) //일반 회복
                     + (bandage[2] * int(rest / bandage[0])), health) //추가 회복
            - attack[1]; //공격
        time = attack[0];
        if(answer <= 0) return -1;
        
    }
    return answer;
}